**Enoncé du problème**

Lors d'un bataille de l'antiquité, les légionnaires constituant l'armée romaine auraient pu se mettre en formation carrée aussi bien qu'en formation triangulaire. Sachant que l'armée compte plus de 10000 hommes, combien de légionnaires romains ont participé à la bataille ?


**Stratégie**

Il est possible de créer des nombres triangulaires et de vérifier s'ils sont également le carré d'un nombre en utilisant la racine carrée.

In [32]:
#Problème des romains

from math import *

#Créer un nombre triangulaire
def triangle(n):
    S=0
    for k in range(n+1):
        S=S+k
    return S


#tester si un nombre est un nombre carré
def carrée(n):
    return int(sqrt(n))==sqrt(n)
        
l=[]
for k in range(10000):
    t=triangle(k)
    if carrée(t): #si le nombre triangulaire t est un nombre carré, il est ajouté à la liste ainsi que son rang
        l.append((k,t))
print(l)

[(0, 0), (1, 1), (8, 36), (49, 1225), (288, 41616), (1681, 1413721), (9800, 48024900)]


**Conclusion**

En testant 10000 nombres triangulaires, on btient la liste de ceux qui sont le carré d'un nombre.

Ainsi, il y en a 3 qui sont inférieurs à 10000 : 1, 36, 1225.
Le premier supérieur à 10000 est 41616 et les deux suivants sont 1 413 721 et 48 024 900.